src/services/services.ts

```diff
@@ -1086,6 +1088,12 @@ namespace ts {
         newText: string;
     }
 
+    export interface TextInsertion {
+        newText: string;
+        /** The position in newText the caret should point to after the insertion. */
+        caretOffset: number;

```

Comment:
It would be interesting if this was actually a range so that the user could just start typing over a default region.